In [89]:
import pandas as pd
import numpy as np
from random import randint
from time import sleep
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

### Khởi tạo driver

In [47]:
driver = webdriver.Chrome(options=options)
driver.maximize_window()

## Truy cập trang bán máy tính bảng của tiki

In [3]:
driver.get("https://tiki.vn/may-tinh-bang/c1794")
sleep(randint(5,10))

#### Lấy link và tên sản phẩm

In [4]:
links_tiki = []
title_tiki = []

In [5]:
while True:
    try:
        products = driver.find_elements(By.CSS_SELECTOR, ".style__ProductLink-sc-7xd6qw-2.fHwskZ.product-item")
        for prod in products:
            links_tiki.append(prod.get_attribute("href"))
            title_tiki.append(prod.find_element(By.CSS_SELECTOR, "h3").text)

        # Load thêm sản phẩm
        more_page = driver.find_element(By.CSS_SELECTOR, ".Pagination__Root-sc-cyke21-0.gNgpAR li:nth-of-type(5) a")
        if more_page.get_attribute("class") == '':
            more_page.click()
            sleep(randint(5,10))
        else:
            break
    except:
        break

In [6]:
len(links_tiki)

92

In [7]:
len(title_tiki)

92

#### Lấy thông tin cụ thể từng sản phẩm

In [8]:
price_tiki = []
discount_price_tiki = []
discount_tiki = []

In [9]:
res_tiki = [] # Độ phân giải
screen_width_tiki = [] # Độ rộng màn hình
os_tiki = [] # Hệ điều hành
cpu_tiki = [] # Chip
cpu_proc_speed_tiki = [] # Tốc độ xử lý cpu
gpu_tiki = [] # Chip gpu
ram_tiki = [] # Dung lượng ram
storage_tiki = [] # Dung lượng lưu trữ
front_camera_tiki = [] # Độ phân giải camera trước
rear_camera_tiki = [] # Độ phân giải camera sau
battery_tiki = [] # Dung lượng pin
size_tiki = [] # Kích thước
weight_tiki = [] # Khối lượng
brand_tiki = [] # Hãng

In [10]:
for link in links_tiki:
    # Truy cập từng link
    driver.get(link)
    
    try:
        # Nếu có discount
        discount_price = int(''.join(driver.find_element(By.CSS_SELECTOR, ".product-price__current-price").text[:-1].split('.')))
        discount = driver.find_element(By.CSS_SELECTOR, ".product-price__discount-rate").text[1:]
        price = int(round(discount_price/(100-int(discount[:-1]))*100,-3))
        price_tiki.append(price)
        discount_price_tiki.append(discount_price)
        discount_tiki.append(discount)
    except:
        try:
            # Không có discount
            discount_price = int(''.join(driver.find_element(By.CSS_SELECTOR, ".product-price__current-price").text[:-1].split('.')))
            price = int(''.join(driver.find_element(By.CSS_SELECTOR, ".product-price__current-price").text[:-1].split('.')))
            discount = "0%"
            price_tiki.append(price)
            discount_price_tiki.append(discount_price)
            discount_tiki.append(discount)
        except:
            # Không có giá
            price_tiki.append(None)
            discount_price_tiki.append(None)
            discount_tiki.append(None)
        
    # Lăn chuột dần dần để hiện thông tin
    height = 0
    while height<=2000:
        driver.execute_script(f'window.scrollTo(0,{height})')
        sleep(1)
        height+=500
       
    # Đến chỗ thông tin sản phẩm
    detail = driver.find_element(By.XPATH, "//div[text()='Thông tin chi tiết']//parent::div//div[2]")
    ActionChains(driver).move_to_element(detail).perform()
    sleep(1)
    
    # Lấy độ phân giải
    try:
        res = detail.find_element(By.XPATH, "//span[text()='Độ phân giải']//parent::div//span[2]").text
        res_tiki.append(res)
    except:
        res_tiki.append(None)
        
    # Lấy độ rộng màn hình    
    try:
        screen_width = detail.find_element(By.XPATH, "//span[text()='Kích thước màn hình']//parent::div//span[2]").text
        screen_width_tiki.append(screen_width)
    except:
        screen_width_tiki.append(None)
        
    # Lấy os    
    try:
        os = detail.find_element(By.XPATH, "//span[text()='Hệ điều hành']//parent::div//span[2]").text
        os_tiki.append(os)
    except:
        os_tiki.append(None)

    # Lấy cpu   
    try:
        cpu = detail.find_element(By.XPATH, "//span[text()='Chip set']//parent::div//span[2]").text
        cpu_tiki.append(cpu)
    except:
        cpu_tiki.append(None)
        
    # Lấy tốc độ cpu    
    try:
        cpu_proc_speed = detail.find_element(By.XPATH, "//span[text()='Tốc độ CPU']//parent::div//span[2]").text
        cpu_proc_speed_tiki.append(cpu_proc_speed)
    except:
        cpu_proc_speed_tiki.append(None)
        
    # Lấy gpu
    try:
        gpu = detail.find_element(By.XPATH, "//span[text()='Chip đồ họa (GPU)']//parent::div//span[2]").text
        gpu_tiki.append(gpu)
    except:
        gpu_tiki.append(None)
    
    # Lấy ram
    try:
        ram = detail.find_element(By.XPATH, "//span[text()='RAM']//parent::div//span[2]").text
        ram_tiki.append(ram)
    except:
        ram_tiki.append(None)
        
    # Lấy dung lượng    
    try:
        storage = detail.find_element(By.XPATH, "//span[text()='ROM']//parent::div//span[2]").text
        storage_tiki.append(storage)
    except:
        storage_tiki.append(None)
    
    # Lấy camera sau
    try:
        rear_camera = detail.find_element(By.XPATH, "//span[text()='Camera sau']//parent::div//span[2]").text
        rear_camera_tiki.append(rear_camera)
    except:
        rear_camera_tiki.append(None)
        
    # Lấy camera trước
    try:
        front_camera = detail.find_element(By.XPATH, "//span[text()='Camera trước']//parent::div//span[2]").text
        front_camera_tiki.append(front_camera)
    except:
        front_camera_tiki.append(None)
    
    # Lấy dung lượng pin
    try:
        battery = detail.find_element(By.XPATH, "//span[text()='Dung lượng pin']//parent::div//span[2]").text
        battery_tiki.append(battery)
    except:
        battery_tiki.append(None)
        
    # Lấy kích thước
    try:
        size = detail.find_element(By.XPATH, "//span[text()='Kích thước']//parent::div//span[2]").text
        size_tiki.append(size)
    except:
        size_tiki.append(None)
        
    # Lấy trọng lượng
    try:
        weight = detail.find_element(By.XPATH, "//span[text()='Trọng lượng sản phẩm']//parent::div//span[2]").text
        weight_tiki.append(weight)
    except:
        weight_tiki.append(None)
        
    # Lấy hãng
    try:
        brand = detail.find_element(By.XPATH, "//span[text()='Thương hiệu']//parent::div//span[2]").text
        brand_tiki.append(brand)
    except:
        brand_tiki.append(None)

## Truy cập trang bán máy tính bảng của cellphones

In [11]:
driver.get("https://cellphones.com.vn/tablet.html")
sleep(randint(5,10))

# Load thêm sản phẩm
while True:
    try:
        view_more = driver.find_element(By.CSS_SELECTOR, ".button.btn-show-more.button__show-more-product")
        ActionChains(driver).move_to_element(view_more).click().perform()
        sleep(randint(5,10))
    except:
        break

#### Lấy danh sách sản phẩm

In [12]:
cellphones_products = driver.find_elements(By.CSS_SELECTOR, ".product-info")

In [13]:
len(cellphones_products)

109

Như vậy có thêm 109 sản phẩm.

#### Lấy link và tên sản phẩm

In [14]:
links_cps = []
title_cps = []

In [15]:
for prod in cellphones_products:
    links_cps.append(prod.find_element(By.CSS_SELECTOR, ".product__link.button__link").get_attribute("href"))
    title_cps.append(prod.find_element(By.CSS_SELECTOR, ".product__name").text)

##### Kiểm tra xem có đủ link và title chưa

In [16]:
len(links_cps)

109

In [17]:
len(title_cps)

109

#### Lấy thông tin về giá

In [18]:
price_cps = [] # Giá gốc
discount_price_cps = [] # Giá sau khi giảm giá
discount_cps = [] # Discount

In [19]:
for prod in cellphones_products:
    try:
        # Nếu có discount
        price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".product__price--through").text[:-1].split('.')))
        discount_price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".product__price--show").text[:-1].split('.')))
        discount = prod.find_element(By.CSS_SELECTOR, ".product__price--percent-detail").text.split(' ')[1]
        price_cps.append(price)
        discount_price_cps.append(discount_price)
        discount_cps.append(discount)
    except:
        try:
            # Không có discount
            price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".product__price--show").text[:-1].split('.')))
            discount_price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".product__price--show").text[:-1].split('.')))
            discount = "0%"
            price_cps.append(price)
            discount_price_cps.append(discount_price)
            discount_cps.append(discount)
        except:
            # Không có giá
            price_cps.append(None)
            discount_price_cps.append(None)
            discount_cps.append(None)

In [20]:
len(price_cps)

109

In [21]:
len(discount_price_cps)

109

In [22]:
len(discount_cps)

109

#### Lấy thông tin cụ thể từng sản phẩm

In [23]:
res_cps = [] # Độ phân giải
screen_width_cps = [] # Độ rộng màn hình
os_cps = [] # Hệ điều hành
cpu_cps = [] # Chip
cpu_proc_speed_cps = [] # Tốc độ xử lý cpu
gpu_cps = [] # Chip gpu
ram_cps = [] # Dung lượng ram
storage_cps = [] # Dung lượng lưu trữ
front_camera_cps = [] # Độ phân giải camera trước
rear_camera_cps = [] # Độ phân giải camera sau
battery_cps = [] # Dung lượng pin
size_cps = [] # Kích thước
weight_cps = [] # Trọng lượng
brand_cps = [] # Hãng

In [24]:
for link in links_cps:
    # Truy cập vào từng link
    driver.get(link)
    
    # Lăn chuột dần dần để hiện thông tin
    height = 0
    while height<=2000:
        driver.execute_script(f'window.scrollTo(0,{height})')
        sleep(1)
        height+=500
    
    # Ấn hiện thêm thông tin
    view_detail = driver.find_element(By.CSS_SELECTOR, ".button.button__show-modal-technical.my-3.is-flex.is-justify-content-center")
    driver.execute_script("arguments[0].click();", view_detail)
    sleep(1)
    
    detail = driver.find_element(By.CSS_SELECTOR, ".technical-content-modal")
    
    # Lấy độ phân giải
    try:
        res = detail.find_element(By.XPATH, "//a[text()='Độ phân giải màn hình']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(res).perform()
        res_cps.append(res.text)
    except:
        res_cps.append(None)
        
    # Lấy độ rộng màn hình    
    try:
        screen_width = detail.find_element(By.XPATH, "//a[text()='Kích thước màn hình']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(screen_width).perform()
        screen_width_cps.append(screen_width.text)
    except:
        screen_width_cps.append(None)
        
    # Lấy os    
    try:
        os = detail.find_element(By.XPATH, "//a[text()='Hệ điều hành']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(os).perform()
        os_cps.append(os.text)
    except:
        os_cps.append(None)

    # Lấy cpu   
    try:
        cpu = detail.find_element(By.XPATH, "//a[text()='Chipset']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(cpu).perform()
        cpu_cps.append(cpu.text)
    except:
        cpu_cps.append(None)
        
    # Lấy tốc độ cpu    
    try:
        cpu_proc_speed = detail.find_element(By.XPATH, "//p[text()='Loại CPU']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(cpu_proc_speed).perform()
        cpu_proc_speed_cps.append(cpu_proc_speed.text)
    except:
        cpu_proc_speed_cps.append(None)
        
    # Lấy gpu
    try:
        gpu = detail.find_element(By.XPATH, "//a[text()='GPU']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(gpu).perform()
        gpu_cps.append(gpu.text)
    except:
        gpu_cps.append(None)
    
    # Lấy ram
    try:
        ram = detail.find_element(By.XPATH, "//a[text()='Dung lượng RAM']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(ram).perform()
        ram_cps.append(ram.text)
    except:
        ram_cps.append(None)
        
    # Lấy dung lượng    
    try:
        storage = detail.find_element(By.XPATH, "//a[text()='Bộ nhớ trong']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(storage).perform()
        storage_cps.append(storage.text)
    except:
        storage_cps.append(None)
    
    # Lấy camera sau
    try:
        rear_camera = detail.find_element(By.XPATH, "//p[text()='Camera sau']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(rear_camera).perform()
        rear_camera_cps.append(rear_camera.text)
    except:
        rear_camera_cps.append(None)
        
    # Lấy camera trước
    try:
        front_camera = detail.find_element(By.XPATH, "//p[text()='Camera trước']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(front_camera).perform()
        front_camera_cps.append(front_camera.text)
    except:
        front_camera_cps.append(None)
    
    # Lấy dung lượng pin
    try:
        battery = detail.find_element(By.XPATH, "//a[text()='Pin']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(battery).perform()
        battery_cps.append(battery.text)
    except:
        battery_cps.append(None)
        
    # Lấy kích thước
    try:
        size = detail.find_element(By.XPATH, "//p[text()='Kích thước']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(size).perform()
        size_cps.append(size.text)
    except:
        size_cps.append(None)
        
    # Lấy trọng lượng
    try:
        weight = detail.find_element(By.XPATH, "//p[text()='Trọng lượng']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(weight).perform()
        weight_cps.append(weight.text)
    except:
        weight_cps.append(None)
        
    # Lấy hãng
    try:
        brand = detail.find_element(By.XPATH, "//p[text()='Hãng']//parent::p//parent::div//div")
        ActionChains(driver).move_to_element(brand).perform()
        brand_cps.append(brand.text)
    except:
        brand_cps.append(None)

## Truy cập trang bán máy tính bảng của viettelstore

In [48]:
driver.get("https://viettelstore.vn/may-tinh-bang")
sleep(randint(5,10))
# Load thêm sản phẩm
while True:
    try:
        view_more = driver.find_element(By.XPATH,"//a[text()='Xem thêm sản phẩm ']")
        ActionChains(driver).move_to_element(view_more).click(view_more).perform()
        sleep(randint(5,10))
    except:
        break

#### Lấy danh sách sản phẩm

In [49]:
viettel_products = driver.find_elements(By.CSS_SELECTOR, ".product-info")

In [50]:
len(viettel_products)

57

Như vây có thêm 57 sản phẩm.

#### Lấy links và tên sản phẩm

In [51]:
links_viettel = []
title_viettel = []

In [52]:
for prod in viettel_products:
    links_viettel.append(prod.find_element(By.CSS_SELECTOR, "a").get_attribute("href"))
    title_viettel.append(prod.find_element(By.CSS_SELECTOR, "h3").text)

##### Kiểm tra xem đủ link và title chưa

In [53]:
len(links_viettel)

57

In [54]:
len(title_viettel)

57

#### Lấy thông tin về giá

In [55]:
price_viettel = [] # Giá gốc
discount_price_viettel = [] # Giá sau khi giảm giá
discount_viettel = [] # Discount

In [56]:
for prod in viettel_products:
    try:
        # Nếu có discount
        price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".price-old").text[:-2].split('.')))
        discount_price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".price").text[:-2].split('.')))
        discount = prod.find_element(By.CSS_SELECTOR, ".percent-discount-text2").text
        price_viettel.append(price)
        discount_price_viettel.append(discount_price)
        discount_viettel.append(discount)
    except:
        try:
            # Không có discount
            price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".price").text[:-2].split('.')))
            discount_price = int(''.join(prod.find_element(By.CSS_SELECTOR, ".price").text[:-2].split('.')))
            discount = "0%"
            price_viettel.append(price)
            discount_price_viettel.append(discount_price)
            discount_viettel.append(discount)
        except:
            # Không có giá
            price_viettel.append(None)
            discount_price_viettel.append(None)
            discount_viettel.append(None)

#### Lấy thông tin cụ thể từng sản phẩm

In [99]:
res_viettel = [] # Độ phân giải
screen_width_viettel = [] # Độ rộng màn hình
os_viettel = [] # Hệ điều hành
cpu_viettel = [] # Chip
cpu_proc_speed_viettel = [] # Tốc độ xử lý cpu
gpu_viettel = [] # Chip gpu
ram_viettel = [] # Dung lượng ram
storage_viettel = [] # Dung lượng lưu trữ
front_camera_viettel = [] # Độ phân giải camera trước
rear_camera_viettel = [] # Độ phân giải camera sau
battery_viettel = [] # Dung lượng pin
size_viettel = [] # Kích thước
weight_viettel = [] # Trọng lượng
brand_viettel = [] # Hãng

In [100]:
for link in links_viettel:
    # Truy cập từng link
    driver.get(link)
    
    # Lăn chuột dần dần để hiện thông tin
    height = 0
    while height<=2000:
        driver.execute_script(f'window.scrollTo(0,{height})')
        sleep(1)
        height+=500
    
    # Ấn hiện thêm thông tin
    try:
        view_detail = driver.find_element(By.CSS_SELECTOR, "#navcau-hinh.detail")
        driver.execute_script("arguments[0].click();", view_detail)
        sleep(1)
    except:
        pass
    
    # Lấy độ phân giải
    try:
        res = driver.find_element(By.XPATH, "//td[text()='Độ phân giải']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(res).perform()
        res_viettel.append(res.text)
    except:
        res_viettel.append(None)
        
    # Lấy độ rộng màn hình    
    try:
        screen_width = driver.find_element(By.XPATH, "//td[text()='Kích thước màn hình']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(screen_width).perform()
        screen_width_viettel.append(screen_width.text)
    except:
        screen_width_viettel.append(None)
        
    # Lấy os    
    try:
        os = driver.find_element(By.XPATH, "//td[text()='Hệ điều hành']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(os).perform()
        os_viettel.append(os.text)
    except:
        os_viettel.append(None)

    # Lấy cpu   
    try:
        cpu = driver.find_element(By.XPATH, "//td[text()='Chipset']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(cpu).perform()
        cpu_viettel.append(cpu.text)
    except:
        cpu_viettel.append(None)
        
    # Lấy tốc độ cpu    
    try:
        cpu_proc_speed = driver.find_element(By.XPATH, "//td[text()='Số nhân CPU']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(cpu_proc_speed).perform()
        cpu_proc_speed_viettel.append(cpu_proc_speed.text)
    except:
        cpu_proc_speed_viettel.append(None)
        
    # Lấy gpu
    try:
        gpu = driver.find_element(By.XPATH, "//td[text()='Chip đồ họa (GPU)']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(gpu).perform()
        gpu_viettel.append(gpu.text)
    except:
        gpu_viettel.append(None)
    
    # Lấy ram
    try:
        ram = driver.find_element(By.XPATH, "//td[text()='RAM']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(ram).perform()
        ram_viettel.append(ram.text)
    except:
        ram_viettel.append(None)
        
    # Lấy dung lượng    
    try:
        storage = driver.find_element(By.XPATH, "//td[text()='Bộ nhớ trong']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(storage).perform()
        storage_viettel.append(storage.text)
    except:
        storage_viettel.append(None)
    
    # Lấy camera sau
    try:
        rear_camera = driver.find_element(By.XPATH, "//td[text()='Camera sau']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(rear_camera).perform()
        rear_camera_viettel.append(rear_camera.text)
    except:
        rear_camera_viettel.append(None)
        
    # Lấy camera trước
    try:
        front_camera = driver.find_element(By.XPATH, "//td[text()='Camera trước']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(front_camera).perform()
        front_camera_viettel.append(front_camera.text)
    except:
        front_camera_viettel.append(None)
    
    # Lấy dung lượng pin
    try:
        battery = driver.find_element(By.XPATH, "//td[text()='Dung lượng Pin']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(battery).perform()
        battery_viettel.append(battery.text)
    except:
        battery_viettel.append(None)
        
    # Lấy kích thước
    try:
        size = driver.find_element(By.XPATH, "//td[text()='Kích thước']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(size).perform()
        size_viettel.append(size.text)
    except:
        size_viettel.append(None)
        
    # Lấy trọng lượng
    try:
        weight = driver.find_element(By.XPATH, "//td[text()='Trọng lượng']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(weight).perform()
        weight_viettel.append(weight.text)
    except:
        weight_viettel.append(None)
        
    # Lấy hãng
    try:
        brand = driver.find_element(By.XPATH, "//td[text()='Hãng']//parent::tr//td[2]")
        ActionChains(driver).move_to_element(brand).perform()
        brand_viettel.append(brand.text)
    except:
        brand_viettel.append(None)

KeyboardInterrupt: 

## Tổng hợp lại

In [ ]:
links = links_viettel + links_cps + links_tiki
title = title_viettel + title_cps + title_tiki
price_list = price_viettel + price_cps + price_tiki
discount_price_list = discount_price_viettel + discount_price_cps + discount_price_tiki
discount_list = discount_viettel + discount_cps + discount_tiki

In [ ]:
res_list = res_viettel + res_cps + res_tiki
screen_width_list = screen_width_viettel + screen_width_cps + screen_width_tiki
os_list = os_viettel + os_cps + os_tiki
cpu_list = cpu_viettel + cpu_cps + cpu_tiki
cpu_proc_speed_list = cpu_proc_speed_viettel + cpu_proc_speed_cps + cpu_proc_speed_tiki
gpu_list = gpu_viettel + gpu_cps + gpu_tiki
ram_list = ram_viettel + ram_cps + ram_tiki
storage_list = storage_viettel + storage_cps + storage_tiki
front_camera_list = front_camera_viettel + front_camera_cps + front_camera_tiki
rear_camera_list = rear_camera_viettel + rear_camera_cps + rear_camera_tiki
battery_list = battery_viettel + battery_cps + battery_tiki
size_list = size_viettel + size_cps + size_tiki
weight_list = weight_viettel + weight_cps + weight_tiki
brand_list = brand_viettel + brand_cps + brand_tiki

In [ ]:
data = pd.DataFrame(list(zip(title, price_list, discount_price_list, discount_list, res_list, screen_width_list, os_list,
                        cpu_list, cpu_proc_speed_list, gpu_list, ram_list, storage_list, front_camera_list, rear_camera_list,
                        battery_list, size_list, weight_list, brand_list , links)), 
                    columns = 'Tên sản phẩm', 'Giá', 'Giá sau khi giảm giá', 'Discount', 'Độ phân giải', 'Độ rộng màn hình',
                   'Hệ điều hành', 'CPU', 'Tốc độ CPU', 'GPU', 'RAM', 'Dung lượng', 'Camere trước', 'Camera sau',
                   'Dung lượng pin', 'Kích thước', 'Trọng lượng', 'Hãng', 'Link')

In [ ]:
data.to_csv("tablet.csv", index=False)